### PR TITLE
[REVIEW] Update Python README with info about DeviceBuffer/MemoryResource and external libraries [skip-ci]

### LINKED ISSUE #446: [DOC] Update Python section of README
The [Python section of the README](https://github.com/rapidsai/rmm#using-rmm-in-python-code) needs to be updated:

* We should remove references to the old `device_array` API. 
* We should add information about the new `DeviceBuffer` and `MemoryResource` APIs
* We should add information about using RMM with other libraries (such as the RMM plugin for Numba)

cc @jakirkham if you have any thoughts/additions here

### PR DESCRIPTION
**Pull request by shwina**
closes #446

### CODE REVIEW

**Comment by shwina** [comment]
cc: @gmarkall @jakirkham would be great to get your input on this. Thanks!

**Comment by shwina** [comment]
rerun tests

File: README.md [hunk 1]
```diff
@@ -307,107 +307,135 @@ RMM does not currently provide any data structures that interface with `host_mem
...
+24
 ```
-def fillna(data, mask, value):
-    out = rmm.device_array_like(data)
-    out.copy_to_device(data)
-    configured = gpu_fill_masked.forall(data.size)
-    configured(value, mask, out)
-    return out
+
+Conversely the data underlying a DeviceBuffer can be copied to the
```
Comment (shwina) [inline README.md:353]:
```suggestion
Conversely, the data underlying a DeviceBuffer can be copied to the
```

**Comment by kkraus14** [comment]
cc @lmeyerovic as I believe you've been asking for doc updates like this

**Comment by gmarkall** [comment]
I see this is already merged, but I've just read through it and the Numba guidance looks clear and correct.

**Comment by jakirkham** [comment]
Yep README looks good. Thanks Ashwin! 😄

**Pull request merged**